compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -634,7 +634,7 @@ object desugar {
       val mdefs = moduleDef(
         ModuleDef(
           className.toTermName, Template(emptyConstructor, parentTpt :: Nil, companionDerived, EmptyValDef, defs))
-            .withMods(companionMods | Synthetic))
+            .withMods(companionMods.withAnnotations(Nil) | Synthetic))
```

Comment:
I would directly change the definition of `companionMods` to not contain annotations (looks like the only other use is for the implicit def generated for implicit classes where not carrying the annotations also makes sense)

Reply:
That makes sense. Done.